test/stdlib/tgmath.swift

```diff
@@ -103,200 +60,135 @@ let gz = CGFloat(3.3)
 
-  (d1, f1, g1) = (cbrt(dx), cbrt(fx), cbrt(gx))
-  expectEqual("cbrt 0.464158883361278 0.464159 cbrt", print3("cbrt", d1, f1, g1))
+  expectEqualWithAccuracy(cbrt(dx), cbrt(fx), cbrt(gx), 0.4641588833612779, ulps: 1)
 
-  (d1, f1, g1) = (sqrt(dx), sqrt(fx), sqrt(gx))
-  expectEqual("sqrt 0.316227766016838 0.316228 sqrt", print3("sqrt", d1, f1, g1))
+  expectEqualWithAccuracy(sqrt(dx), sqrt(fx), sqrt(gx), 0.31622776601683794, ulps: 1)
 
-  (d1, f1, g1) = (erf(dx), erf(fx), erf(gx))
-  expectEqual("erf 0.112462916018285 0.112463 erf", print3("erf", d1, f1, g1))
+  expectEqualWithAccuracy(erf(dx), erf(fx), erf(gx), 0.1124629160182849, ulps: 1)
 
-  (d1, f1, g1) = (erfc(dx), erfc(fx), erfc(gx))
-  expectEqual("erfc 0.887537083981715 0.887537 erfc", print3("erfc", d1, f1, g1))
+  expectEqualWithAccuracy(erfc(dx), erfc(fx), erfc(gx), 0.8875370839817152, ulps: 1)
 
-  (d1, f1, g1) = (tgamma(dx), tgamma(fx), tgamma(gx))
-  expectEqual("tgamma 9.51350769866873 9.51351 tgamma",
-    print3("tgamma", d1, f1, g1))
+  expectEqualWithAccuracy(tgamma(dx), tgamma(fx), tgamma(gx), 9.51350769866873, ulps: 1)
 
-  (d1, f1, g1) = (ceil(dx), ceil(fx), ceil(gx))
-  expectEqual("ceil 1.0 1.0 ceil", print3("ceil", d1, f1, g1))
+  expectEqualWithAccuracy(ceil(dx), ceil(fx), ceil(gx), 1.0, ulps: 0)
 
-  (d1, f1, g1) = (floor(dx), floor(fx), floor(gx))
-  expectEqual("floor 0.0 0.0 floor", print3("floor", d1, f1, g1))
+  expectEqualWithAccuracy(floor(dx), floor(fx), floor(gx), 0.0, ulps: 0)
 
-  (d1, f1, g1) = (nearbyint(dx), nearbyint(fx), nearbyint(gx))
-  expectEqual("nearbyint 0.0 0.0 nearbyint", print3("nearbyint", d1, f1, g1))
+  expectEqualWithAccuracy(nearbyint(dx), nearbyint(fx), nearbyint(gx), 0.0, ulps: 0)
 
-  (d1, f1, g1) = (rint(dx), rint(fx), rint(gx))
-  expectEqual("rint 0.0 0.0 rint", print3("rint", d1, f1, g1))
+  expectEqualWithAccuracy(rint(dx), rint(fx), rint(gx), 0.0, ulps: 0)
 
-  (d1, f1, g1) = (round(dx), round(fx), round(gx))
-  expectEqual("round 0.0 0.0 round", print3("round", d1, f1, g1))
+  expectEqualWithAccuracy(round(dx), round(fx), round(gx), 0.0, ulps: 0)
 
-  (d1, f1, g1) = (trunc(dx), trunc(fx), trunc(gx))
-  expectEqual("trunc 0.0 0.0 trunc", print3("trunc", d1, f1, g1))
+  expectEqualWithAccuracy(trunc(dx), trunc(fx), trunc(gx), 0.0, ulps: 0)
 }
 
 MathTests.test("Binary functions") {
+    expectEqualWithAccuracy(atan2(dx, dy), atan2(fx, fy), atan2(gx, gy), 0.04542327942157701, ulps: 1)
+    expectEqualWithAccuracy(hypot(dx, dy), hypot(fx, fy), hypot(gx, gy), 2.202271554554524, ulps: 1)
+    expectEqualWithAccuracy(pow(dx, dy), pow(fx, fy), pow(gx, gy), 0.00630957344480193, ulps: 1)
+    expectEqualWithAccuracy(fmod(dx, dy), fmod(fx, fy), fmod(gx, gy), 0.1, ulps: 0)
+    expectEqualWithAccuracy(remainder(dx, dy), remainder(fx, fy), remainder(gx, gy), 0.1, ulps: 0)
+    expectEqualWithAccuracy(copysign(dx, dy), copysign(fx, fy), copysign(gx, gy), 0.1, ulps: 0)
+
+    expectEqual(nextafter(dx, dy), 0.10000000000000002)
```

Comment:
This should be written `expectEqual(nextafter(dx, dy), dx.nextUp)`. This is both more precise and also lets you eliminate the `arch` condition below.

Reply:
Good idea!  I've also added a check that `nextafter()` in the down direction matches `.nextDown`